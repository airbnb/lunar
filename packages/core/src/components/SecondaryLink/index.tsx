import React from 'react';
import Link, { Props } from '../Link';

const StyledSecondaryLink = Link.extendStyles(({ color }) => ({
  link: {
    color: color.core.neutral[5],

    ':hover': {
      color: color.core.neutral[6],
    },
  },
}));

/** A link to use for secondary actions. Supports all the same props as `Link`. */
export default function SecondaryLink({ children, ...props }: Props) {
  return <StyledSecondaryLink {...props}>{children}</StyledSecondaryLink>;
}
