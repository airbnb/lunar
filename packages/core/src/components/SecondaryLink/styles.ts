import Core from '../..';
import { StyleSheet } from '../../hooks/useStyles';
import baseStyleSheet from '../Link/styles';

const styleSheet: StyleSheet = Core.aesthetic.extendStyles(baseStyleSheet, ({ color }) => ({
  link: {
    color: color.core.neutral[5],

    ':hover': {
      color: color.core.neutral[6],
    },
  },
}));

export default styleSheet;
