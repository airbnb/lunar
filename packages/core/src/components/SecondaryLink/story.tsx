import React from 'react';
import IconAddAlt from '@airbnb/lunar-icons/lib/interface/IconAddAlt';
import SecondaryLink from '.';

export default {
  title: 'Core/SecondaryLink',
  parameters: {
    inspectComponents: [SecondaryLink],
  },
};

export function aSecondaryLink() {
  return <SecondaryLink href="https://github.com/airbnb/lunar">Link</SecondaryLink>;
}

aSecondaryLink.story = {
  name: 'A secondary link.',
};

export function aButtonLink() {
  return <SecondaryLink>Link</SecondaryLink>;
}

aButtonLink.story = {
  name: 'A button link.',
};

export function withDifferentSizingSmallRegularDefaultAndLarge() {
  return (
    <>
      <SecondaryLink small href="https://github.com/airbnb/lunar">
        Link
      </SecondaryLink>{' '}
      <SecondaryLink href="https://github.com/airbnb/lunar">Link</SecondaryLink>{' '}
      <SecondaryLink large href="https://github.com/airbnb/lunar">
        Link
      </SecondaryLink>
    </>
  );
}

withDifferentSizingSmallRegularDefaultAndLarge.story = {
  name: 'With different sizing: small, regular (default), and large.',
};

export function withDifferentStatesMutedDisabledAndInverted() {
  return (
    <>
      <SecondaryLink muted href="https://github.com/airbnb/lunar">
        Link
      </SecondaryLink>{' '}
      <SecondaryLink disabled href="https://github.com/airbnb/lunar">
        Link
      </SecondaryLink>{' '}
      <SecondaryLink inverted href="https://github.com/airbnb/lunar">
        Link
      </SecondaryLink>
    </>
  );
}

withDifferentStatesMutedDisabledAndInverted.story = {
  name: 'With different states: muted, disabled, and inverted.',
};

export function withBeforeAndOrAfterIcons() {
  return (
    <>
      <SecondaryLink href="https://github.com/airbnb/lunar" beforeIcon={<IconAddAlt decorative />}>
        Link
      </SecondaryLink>
      <br />
      <SecondaryLink href="https://github.com/airbnb/lunar" afterIcon={<IconAddAlt decorative />}>
        Link
      </SecondaryLink>
    </>
  );
}

withBeforeAndOrAfterIcons.story = {
  name: 'With before and or after icons.',
};

export function stopClickAndDefaultEventsWhenDisabled() {
  return (
    <SecondaryLink disabled href="https://github.com/airbnb/lunar" onClick={action('onClick')}>
      Link
    </SecondaryLink>
  );
}

stopClickAndDefaultEventsWhenDisabled.story = {
  name: 'Stop click and default events when disabled.',
};

export function boldText() {
  return (
    <SecondaryLink bold href="https://github.com/airbnb/lunar">
      Link
    </SecondaryLink>
  );
}

boldText.story = {
  name: 'Bold text.',
};
