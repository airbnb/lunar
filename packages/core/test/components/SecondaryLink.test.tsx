import React from 'react';
import { shallow } from 'enzyme';
import { shallowWithStyles } from '@airbnb/lunar-test-utils';
import SecondaryLink from '../../src/components/SecondaryLink';

describe('<SecondaryLink />', () => {
  it('renders a secondary link', () => {
    const wrapper = shallowWithStyles(shallow(<SecondaryLink>Link</SecondaryLink>).getElement());

    expect(wrapper).toMatchSnapshot();
  });
});
