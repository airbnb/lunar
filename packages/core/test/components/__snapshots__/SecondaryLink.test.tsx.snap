// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SecondaryLink /> renders a secondary link 1`] = `
<Text
  baseline={false}
  bold={false}
  inline={true}
  large={false}
  small={false}
>
  <ButtonOrLink
    className="link_ebyk5g"
  >
    Link
  </ButtonOrLink>
</Text>
`;
